GithubRepo/Data/Repository/GithubMainRepository.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  GithubMainRepository.swift
+//  GithubRepo
+//
+//  Created by Marcos Alves on 15/10/21.
+//
+
+import Foundation
+
+class GithubMainRepository: GithubRepository {
+    
+    weak var delegate: GithubRepositoryDelegate?
+    private var dataSource: GithubDataSource
+    private(set) var repositories: [Repository]
+    private(set) var loading: Bool
+    private(set) var error: Bool
+    
+    init(dataSource: GithubDataSource = RemoteGithubDataSource()) {
+        self.dataSource = dataSource
+        self.repositories = []
+        self.loading = false
+        self.error = false
+    }
+    
+    func handleChangeRepositories(_ repo: [RepositoryResponse]) {
```

Comment:
Helper functions should be defined as `private`.

Usually, the `public` methods are the ones from the `protocol`